server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -609,25 +610,24 @@ private void updateShard(DiscoveryNodes nodes, ShardRouting shardRouting, Shard
                 .collect(Collectors.toSet());
             shard.updateShardState(shardRouting, primaryTerm, primaryReplicaSyncer::resync, clusterState.version(),
                 inSyncIds, indexShardRoutingTable, pre60AllocationIds);
-        } catch (Exception e) {
-            failAndRemoveShard(shardRouting, true, "failed updating shard routing entry", e, clusterState);
-            return;
-        }
 
-        final IndexShardState state = shard.state();
-        if (shardRouting.initializing() && (state == IndexShardState.STARTED || state == IndexShardState.POST_RECOVERY)) {
-            // the master thinks we are initializing, but we are already started or on POST_RECOVERY and waiting
-            // for master to confirm a shard started message (either master failover, or a cluster event before
-            // we managed to tell the master we started), mark us as started
-            if (logger.isTraceEnabled()) {
-                logger.trace("{} master marked shard as initializing, but shard has state [{}], resending shard started to {}",
-                    shardRouting.shardId(), state, nodes.getMasterNode());
-            }
-            if (nodes.getMasterNode() != null) {
-                shardStateAction.shardStarted(shardRouting, "master " + nodes.getMasterNode() +
-                        " marked shard as initializing, but shard state is [" + state + "], mark shard as started",
-                    SHARD_STATE_ACTION_LISTENER, clusterState);
+            final IndexShardState state = shard.state();
+            if (shardRouting.initializing() && (state == IndexShardState.STARTED || state == IndexShardState.POST_RECOVERY)) {
+                // the master thinks we are initializing, but we are already started or on POST_RECOVERY and waiting
+                // for master to confirm a shard started message (either master failover, or a cluster event before
+                // we managed to tell the master we started), mark us as started
+                if (logger.isTraceEnabled()) {
+                    logger.trace("{} master marked shard as initializing, but shard has state [{}], resending shard started to {}",
+                        shardRouting.shardId(), state, nodes.getMasterNode());
+                }
+                if (nodes.getMasterNode() != null) {
+                    shardStateAction.shardStarted(shardRouting, primaryTerm, "master " + nodes.getMasterNode() +
+                            " marked shard as initializing, but shard state is [" + state + "], mark shard as started",
+                        SHARD_STATE_ACTION_LISTENER, clusterState);
+                }
             }
+        } catch (Exception e) {
```

Comment:
why change the exception handling here?
While we except an exception to be thrown at times by `createShard`, the same does not hold for the `shardStarted` event that was wrapped here.

Reply:
I think I mixed up things while cherry picking an older commit :/